hi-ml-azure/src/health/azure/azure_util.py

```diff
@@ -513,13 +714,104 @@ def get_aml_runs(args: Namespace, workspace: Workspace, run_id_source: AzureRunI
     elif run_id_source == AzureRunIdSource.RUN_ID:
-        runs = get_aml_runs_from_runids(args, workspace)
+        runs = [get_aml_run_from_run_id_args(args, workspace)]
+    elif run_id_source == AzureRunIdSource.RUN_IDS:
+        runs = get_aml_runs_from_run_ids(args, workspace)
     else:
         raise ValueError(f"Unrecognised RunIdSource: {run_id_source}")
     return [run for run in runs if run is not None]
+
+
+def get_run_file_names(run: Run, prefix: str = "") -> List[str]:
+    """
+    Get the remote path to all files for a given Run which optionally start with a given prefix
+
+    :param run: The AML Run to look up associated files for
+    :param prefix: The optional prefix to filter Run files by
+    :return: A list of paths within the Run's container
+    """
+    all_files = run.get_file_names()
+    return [f for f in all_files if f.startswith(prefix)] if prefix else all_files
+
+
+def download_run_files(run: Run, output_dir: Path, prefix: str = "") -> None:
+    """
+    Download all files for a given run, which optionally start with a given prefix
+
+    :param run: The AML Run to download associated files for
+    :param output_dir: Local directory to which the Run files should be downloaded.
+    :param prefix: Optional prefix to filter Run files by
+    """
+    run_paths = get_run_file_names(run, prefix=prefix)
+    if len(run_paths) == 0:
+        raise ValueError("No such files were found for this Run.")
+
+    for run_path in run_paths:
+        output_path = output_dir / run_path
+        download_run_file(run, run_path, output_path)
+
+
+def download_run_files_from_run_id(run_id: str, output_dir: Path, prefix: str = "",
+                                   workspace: Optional[Workspace] = None,
+                                   workspace_config_path: Optional[Path] = None) -> None:
+    """
+    For a given Azure ML run id, first retrieve the Run, and then download all files,
+    which optionally start with a given prefix
+
+    :param run_id: The id of the Azure ML Run
+    :param output_dir: Local directory to which the Run files should be downloaded.
+    :param prefix: Optional prefix to filter Run files by
+    :param workspace: Optional Azure ML Workspace object
+    :param workspace_config_path: Optional path to settings for Azure ML Workspace
+    """
+    workspace = get_workspace(aml_workspace=workspace, workspace_config_path=workspace_config_path)
+    run = get_aml_run_from_run_id(run_id, aml_workspace=workspace)
+    download_run_files(run, output_dir, prefix=prefix)
+
+
+def download_run_file(run: Run, filename: str, output_path: Path, validate_checksum: bool = False) -> None:
+    """
+    A wrapper around AML Run's download_file method, that handles timeouts
+
+    :param run: The AML Run to download associated file for
+    :param filename: The name of the file as it exists in Azure storage
+    :param output_path: Local path to which the file should be downloaded
```

Comment:
should output_path be a folder or a file? Can we handle both? Should we return the Path to which the file was actually downloaded?

Reply:
This function is for downloading a single file so I think output_path should be a  single file. Yes I'll edit it 
 return the path